testing/file_test/testdata/unattached_multi_file.carbon

```diff
@@ -6,12 +6,11 @@
 // CHECK:STDERR: unattached message 3
 // CHECK:STDERR: unattached message 4
 
+// CHECK:STDOUT: 3 args: `default_args`, `a.carbon`, `b.carbon`
+// CHECK:STDOUT: unattached message 1
+// CHECK:STDOUT: unattached message 2
```

Comment:
I think we previously wanted to put these lines at the end of the overall test file, because they would be expected to describe the output of the overall test. Does that not work out with the current plan for multi-file output checking? (I don't have a strong opinion, if it's too hard or problematic to include them later.)

Reply:
This is what I was trying to point out at https://github.com/carbon-language/carbon-lang/pull/3184#discussion_r1316265323, that starting with file_number=0 may not yield the right behavior. But really, we should consider #3219 and whether change this would/should have an impact on tests.